File: Fila/Fila.c
#include <stdio.h>
#include <stdlib.h>

int fila[5]; //fila[0], fila[1], fila[2], fila[3], fila[4]
int inicio = 0;
int fim = -1;

int mostrarMenu() { //Mostra o Menu na Tela
	int opc;
	
	printf("--------------- MENU ---------------\n\n");
	printf("[1] - Inserir um elemento na Fila\n");
	printf("[2] - Remover um elemento na Fila\n");
	printf("[3] - Verificar condicao da Fila\n");
	printf("[0] - Encerrar o Programa\n\n");
	printf("------------------------------------\n\n");
	printf("Digite sua escolha: ");
	scanf("%d",&opc);
	system("cls");
	
	return opc;
}

void inserirFila(int e) { //Insere um Elemento na Fila
	if (fim == 4) {
		printf("\n!!! A FILA ESTA CHEIA !!!\n\nNao eh possivel inserir mais elementos.");
	}
	else {
		fim = fim + 1; //fim++
		fila[fim] = e;
		printf("\nElemento %d foi inserido na fila.", e);
	}
	
}

void removerFila() { //Remove um Elemento na FIla
	int elementoRemovido, i;
	//int aux;
	
	if (fim == -1){
		printf("!!! A FILA ESTA VAZIA !!!\n\nNao eh possivel remover mais elementos.");
	}
	else {
		elementoRemovido = fila[inicio];
		for (i = 1 ; i <= fim ; i++){
			 //aux = fila[i];
			 //fila[i] = fila[i+1];
			 //fila[i-1] = aux;
			 fila[i-1] = fila[i];
		}
		fim = fim - 1;
		
		printf("O elemento %d foi removido da fila.", elementoRemovido);
	}
}

void verificarCondicaoFila() { //Imprime a Fila atual
	int i;
	
	if (fim == 4){
		printf("!!! A FILA ESTA CHEIA !!!\n\n");
	}
	if (fim == -1){
		printf("!!! A FILA ESTA VAZIA !!!");
	}
	if (fim != -1){
		printf("A sua fila atual eh: \n\n");
		printf("FILA = ");
		for (i = 0 ; i < fim+1 ; i++){
			printf("- [ %d ] ", fila[i]);
		}
	}
} 

void sairPrograma() { //Mostra a Saída na Tela
	printf("--------------- SAIDA ---------------\n\n");
	printf("Encerrando o Programa...\n\n");
    printf("-------------------------------------\n\n");

    system("pause");
}

int main(int argc, char *argv[]) {
	int opc, elemento;

	do {
		opc = mostrarMenu();
		
		if (opc >= 4 || opc < 0) {
			printf("OPCAO INVALIDA - Escolha uma nova opcao. \n\n");
		}
		
		if (opc == 1){
			printf("--------------- ENTRADA ---------------\n\n");
			printf("Digite um elemento para entrar na fila: ");
			scanf("%d",&elemento);

			inserirFila(elemento);

			printf("\n\n");
			printf("---------------------------------------\n\n");
			system("pause");
			system("cls");
		}
		
		if (opc == 2){
			printf("--------------- REMOCAO ---------------\n\n");

			removerFila();

			printf("\n\n");
			printf("---------------------------------------\n\n");
			system("pause");
			system("cls");	
		}
		
		if (opc == 3){
			printf("--------------- CONDICAO ---------------\n\n");

			verificarCondicaoFila();

			printf("\n\n");
			printf("----------------------------------------\n\n");
			system("pause");
			system("cls");
		}
		
		if (opc == 0) {
			sairPrograma();
		}
	} while(opc != 0);	
	return 0;
}

File: Fila/Fila.py
class Fila:
    def __init__(self):
        self.inicio = 0
        self.fim = -1
        self.itens = [0] * 5

def mostrar_menu():
    print("------------------- MENU -------------------\n")
    print("[1] - Inserir um elemento na Fila")
    print("[2] - Remover um elemento da Fila")
    print("[3] - Verificar condição da Fila")
    print("[0] - Encerrar o Programa\n")
    print("--------------------------------------------\n")
    opc = int(input("DIGITE SUA ESCOLHA: "))
    print()
    return opc

def inserir_fila(e, f):
    f.fim += 1
    f.itens[f.fim] = e
    print("\nElemento", e, "foi inserido na Fila.\n\n\n")

def remover_fila(f):
    if f.fim == -1:
        print("!!! A Fila ESTA VAZIA !!!\n")
        print("Não é possível remover mais elementos.\n\n\n")
    else:
        elemento_removido = f.itens[f.inicio]
        for i in range(1, f.fim + 1):
            f.itens[i - 1] = f.itens[i]
        f.fim -= 1
        print("O elemento", elemento_removido, "foi removido da Fila.\n\n\n")

def verificar_condicao_fila(f):
    if f.fim == 4:
        print("!!! A FILA ESTA CHEIA !!!\n")
        print("FILA =", end=" ")
        for i in range(f.fim + 1):
            print("- [", f.itens[i], "]", end=" ")
        print("\n\n\n")
    elif f.fim == -1:
        print("!!! A FILA ESTA VAZIA !!!\n\n\n")
    elif f.fim != -1:
        print("A sua Fila atual é:\n")
        print("FILA =", end=" ")
        for i in range(f.fim + 1):
            print("- [", f.itens[i], "]", end=" ")
        print("\n\n\n")

f = Fila()

while True:
    opc = mostrar_menu()
    
    if opc >= 4 or opc < 0:
        print("OPCAO INVALIDA - Escolha uma nova opcao.\n\n")
    
    if opc == 1:
        print("----------------- ENTRADA -----------------\n")
        if f.fim != 4:
            elemento = int(input("Digite um elemento para entrar na Fila: "))
            inserir_fila(elemento, f)
        else:
            print("!!! A Fila ESTA CHEIA !!!\n\nNão é possível inserir mais elementos.\n\n\n")

    if opc == 2:
        print("----------------- REMOCAO -----------------\n")
        
        remover_fila(f)
    
    if opc == 3:
        print("----------------- CONDICAO -----------------\n")
        
        verificar_condicao_fila(f)
    
    if opc == 0:
        print("------------------ SAIDA ------------------\n")
        print("Encerrando o Programa...            =)\n")
        input("-------------------------------------------\n\n")
        break
